### PR TITLE
add rich information about betterttv emotes (including a thumbnail)

### LINKED ISSUE #18: Add thumbnails for betterttv emote links
I would like for links like `https://betterttv.com/emotes/<hexadecimal emote id>` to have thumbnails to see the emotes, the emote image address is `https://cdn.betterttv.net/emote/<hexadecimal emote id>/3x`

## Example
bttv emote link: https://betterttv.com/emotes/5c07ca765f63c823e97fe4d2
image link: https://cdn.betterttv.net/emote/5c07ca765f63c823e97fe4d2/3x

### PR DESCRIPTION
**Pull request by pajlada**
Fixes #18

### CODE REVIEW

File: link_resolver_betterttv.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+			Status:    200,
+			Tooltip:   tooltip.String(),
+			Thumbnail: thumbnailURL,
+		}, nil, noSpecialDur
+	}
+
+	cache := newLoadingCache("betterttv_emotes", load, 1*time.Hour)
+	emotePathRegex := regexp.MustCompile(`/emotes/([a-f0-9]+)`)
+
+	// Find clips that look like https://clips.twitch.tv/SlugHere
```
Comment (RAnders00) [inline link_resolver_betterttv.go:131]: comment should be updated

File: link_resolver_betterttv.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	customURLManagers = append(customURLManagers, customURLManager{
+		check: func(url *url.URL) bool {
+			if !strings.HasSuffix("."+url.Host, ".betterttv.com") {
```
Comment (RAnders00) [inline link_resolver_betterttv.go:134]: Maybe put a small comment explaining this magic, I remember you giving a short explanation of why you used this

**Comment by alazymeme** [comment]
Is there any chance we could fetch a smaller version of the emote? The current implementation has the emote quite large:
image

**Comment by gempir** [comment]
That's more of an issue of Chatterino not the API. Because Chatterino will display it as any link thumbnail.

**Comment by alazymeme** [comment]
> That's more of an issue of Chatterino not the API. Because Chatterino will display it as any link thumbnail.

This PR points to the 3x version of the emote. Can this not be fixed by simply pointing it to the 2x/1x image?

https://github.com/Chatterino/api/blob/694bef7b07f195a492468a41cfdbca3a3545574f/link_resolver_betterttv.go#L37

**Comment by gempir** [comment]
No. Chatterino will scale the image anyway to the size defined in the Hover Link Thumbail Option

File: link_resolver_betterttv.go (hunk 2 above)
Comment (pajlada) [reply link_resolver_betterttv.go:141]: The idea was to ensure we would match both www.betterttv.com and betterttv.com.
Writing this down in a comment made me realize that we would also be matching kkona.betterttv.com which is not something we wanted to do, so I replaced this with a simpler "golang set" of domains

**Comment by pajlada** [comment]
Looks good to me! :+1:

**Comment by RAnders00** [comment]
> Looks good to me! 👍

image

**Pull request merged**